fish-rust/src/abbrs.rs

```diff
@@ -28,7 +28,7 @@ pub fn abbrs_get_set() -> MutexGuard<'static, AbbreviationSet> {
     ABBRS.lock().unwrap()
 }
 
-/// Controls where in the command line abbreviations may expand.
+// Controls where in the command line abbreviations may expand.
```

Comment:
> They confuse the test drivers

How? Doc comments should be standard Rust so I'm surprised they cause problems.

Reply:
Most of them were fine, but there are a few that actually look like tests, but were either not syntactic Rust:

https://github.com/fish-shell/fish-shell/blob/29bd6eebd038a2984ac90ad5b4e6b3c8a88affed/fish-rust/src/tokenizer.rs#L886
(I'm not sure why this one was picked up as a code block)

Or didn't compile:

https://github.com/fish-shell/fish-shell/blob/29bd6eebd038a2984ac90ad5b4e6b3c8a88affed/fish-rust/src/common.rs#L1793-L1804

Reply:
So it looks like turning the crate from `staticlib` to `rlib` enabled `cargo test --doc`. Interesting.

Searching for

    /// ```

shows only twelve occurrences.

Additionally there are like one or two `///    ` in a new paragraph (probably it uses Markdown rules for code blocks).

I think we should rather fix those few places by removing the backticks, and for the pipe one delete the blank line or unindent it.

Doc comments are not super important but they can be nice (in LSP hover info, completion labels etc). Doc tests are also nice because they show compile-checked usage examples.

Reply:
Ah, I didn't know that's where they showed up! OK, I will do that instead.

Reply:
I think let's just disable them for now. That will speed things up too.

Reply:
Code blocks in doc comments that shouldn't become doctests can be tagged accordingly: https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html#attributes